db/fetcher/indexer_iterators.go

```diff
@@ -501,38 +431,176 @@ func createIndexIterator(
 			}
 			valArr = append(valArr, valueBytes)
 		}
-		if op == opIn {
-			var iter filterValueIndexIterator
-			if isUnique {
-				iter = &eqSingleIndexIterator{
-					indexKey: indexDataStoreKey,
-					execInfo: execInfo,
-				}
-			} else {
-				iter = &eqPrefixIndexIterator{
-					indexKey: indexDataStoreKey,
-					execInfo: execInfo,
-				}
+		return newNinIndexCmp(valArr, op == opIn), nil
+	case opLike, opNlike:
+		return newLikeIndexCmp(filterVal.(string), op == opLike), nil
+	case opAny:
+		return &anyMatcher{}, nil
+	}
+
+	return nil, errors.New("invalid index filter condition")
```

Comment:
todo: Please declare this error in the errors.go file :)

Reply:
done